scripts/wsrep_sst_clone.sh

```diff
@@ -0,0 +1,1155 @@
+        #we will try to silently shutdown the instance
+        `$MYSQL_ACLIENT -e "SHUTDOWN" 2> /dev/null` || :
+        exit 1
+    fi
+
+    # Wait for the receiver process to start
+    until [ -n "$(cat $CLONE_PID_FILE 2>/dev/null)" ]
+    do
+        sleep 0.2
+    done
+    CLONE_REAL_PID=`cat $CLONE_PID_FILE`
+
+    if [[ -n "$WSREP_LOG_DEBUG" ]]; then
+        GRANTS=`$MYSQL_ACLIENT -uroot -NB -e "show grants for '$CLONE_USER'@'localhost'"`
+        wsrep_log_debug "-> Clone user grants: $GRANTS"
+    fi
+
+    trap cleanup_joiner EXIT
+
+    export MYSQL_PWD=$CLONE_PSWD
+    wsrep_log_debug "-> Exported MySQL password $MYSQL_PWD"
+
+    wsrep_log_info "Waiting for clone recipient daemon to be ready for connections at port $JOINER_CLONE_PORT"
+    wsrep_log_debug "-> connecting as: $MYSQL_CLIENT -u$CLONE_USER -h$JOINER_CLONE_HOST -P$JOINER_CLONE_PORT"
+    to_wait=$JOINER_TIMEOUT_WAIT_CLONE_INSTANCE # 90 sec by defaults but it can be changed in the config
+    until $MYSQL_CLIENT -u$CLONE_USER -h$JOINER_CLONE_HOST -P$JOINER_CLONE_PORT \
+          -e 'SELECT USER()' > /dev/null
+    do
+        if [ $to_wait -eq 0 ]
+        then
+            wsrep_log_error "Timeout waiting for clone recipient daemon"
+            kill -9 $CLONE_REAL_PID || :
+            exit $ETIMEDOUT
+        fi
+        to_wait=$(( $to_wait - 1 ))
+        sleep 1
+    done
+
+    wsrep_log_info "Joiner setup."
+    setup_clone_plugin "recipient"
+
+    wsrep_log_info "Waiting for clone recipient daemon to finish"
+
+   # Getting clone process status report
+    OLD_STATUS=""
+    while check_pid_file "$CLONE_PID_FILE"
+    do
+        STATUS=`$MYSQL_ACLIENT -NB -e "select format(((data/estimate)*100),2) 'completed%' from performance_schema.clone_progress where stage like 'FILE_COPY';" 2> /dev/null` || :
+        if [ "$OLD_STATUS" != "$STATUS" ]; then
+            wsrep_log_info "Copy at: ${STATUS}%"
+            OLD_STATUS=$STATUS
+        fi
+        #wsrep_log_debug "Checking parent: $WSREP_SST_OPT_PARENT"
+        check_parent $WSREP_SST_OPT_PARENT
+        sleep 1 # Should this be in the config as well?
+    done
+
+    CLONE_REAL_PID=0
+    wsrep_log_info "clone recepient daemon finished"
+
+    # Execute first restart for
+    wsrep_log_info "FIRST RESTART to fix the dictionary"
+    wsrep_log_info "Performing data recovery"
+    wsrep_log_debug "-> RECOVERY COMMAND LINE: $CLONE_ENV $CLONE_BINARY $DEFAULT_OPTIONS --wsrep_provider=none"
+
+    # Instance should have been gone, but we check the p
+    if [ -s "$CLONE_PID_FILE" ]; then
+        wsrep_log_debug "Cleaning leftover from previous mysqld run $CLONE_PID_FILE"
+        kill -9 `cat $CLONE_PID_FILE` || true
+        rm -f $CLONE_PID_FILE || true
+    fi
+
+# Remove created clone user
+cat << EOF > "$CLONE_SQL"
+SET SESSION sql_log_bin=OFF;
+DROP USER IF EXISTS $CLONE_USER;
+DROP USER IF EXISTS $CLONE_USER@'localhost';
```

Comment:
Solution for [THIS](https://github.com/percona/percona-xtradb-cluster/pull/1961/files#r1886504693) problem:
`mysql.pxc.sst.user` has been cloned from the donor side (this is the side effect). We don't want to have this user on the joiner side. We need to drop it here. 
Please add `DROP USER IF EXISTS 'mysql.pxc.sst.user'@localhost;`
It will cause cleanup of `mysql.role_edges` and 'mysql.default_roles` tables.

Reply:
done